tests/model_formsets/tests.py

```diff
@@ -451,6 +451,41 @@ def test_min_num_with_existing(self):
         formset = AuthorFormSet(queryset=qs)
         self.assertEqual(len(formset.forms), 1)
 
+    def test_edit_only_prevent_instance_creation(self):
+        Author.objects.create(name='Charles Baudelaire')
+        AuthorFormSet = modelformset_factory(Author, fields="__all__")
+        data = {
+            'form-TOTAL_FORMS': '1',
+            'form-INITIAL_FORMS': '0',  # would be 1 normally, but pretend the user edited it.
+            'form-MAX_NUM_FORMS': '0',
+            'form-0-name': 'Walt Whitman',
+        }
+        formset = AuthorFormSet(data, queryset=Author.objects.all(), edit_only=True)
+        self.assertTrue(formset.is_valid())
+        formset.save()
+        self.assertQuerysetEqual(Author.objects.all(), [
+            '<Author: Charles Baudelaire>',
+        ])
+
+    def test_instance_creation_without_edit_only(self):
```

Comment:
Move this test to be first (in the change here). Then we're testing "without edit_onlt", then "with edit_only". 


Remove the comments, I don't think they add too much.